dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/dispatcher/internal/channel/HttpDispatcherLink.java

```diff
@@ -249,6 +251,11 @@ public void close(VirtualConnection conn, Exception e) {
                 this.myChannel.decrementActiveConns();
             }
         }
+
+        if (this.vc != null && this.vc.getStateMap().get(TransportConstants.UPGRADED_CONNECTION) != null) {
+            CountDownLatch countDownLatch = (CountDownLatch) (this.vc.getStateMap().get(TransportConstants.ON_CLOSE_COUNTDOWN_LATCH));
+            countDownLatch.countDown();
```

Comment:
Should i have a countDownLatch != null check just in case?

Reply:
They're not really necessary, but null checks would mirror how the statemap is typically used